chain/types/tx_signing.go

```diff
@@ -114,7 +96,9 @@ func (s ReimbursementTxSigner) SignTx(tx *Transaction, prv *ecdsa.PrivateKey) (*
 	if err != nil {
 		return nil, err
 	}
-	return tx.WithSignature(s, sig)
+	cpy := &Transaction{data: tx.data, gasPayer: tx.gasPayer}
+	cpy.data.Sig = sig
```

Comment:
As weak as `DefaultSigner.SignTx`. Maybe we need a `Copy` function for `Transaction`